checkov/arm/checks/resource/AKSApiServerAuthorizedIpRanges.py

```diff
@@ -25,11 +27,18 @@ def scan_resource_conf(self, conf):
                             return CheckResult.PASSED
             else:
                 # ApiServerAuthorizedIpRanges fully supported in all future API versions
-                if "properties" in conf:
-                    if "apiServerAccessProfile" in conf["properties"]:
-                        if "authorizedIPRanges" in conf["properties"]["apiServerAccessProfile"]:
-                            if conf["properties"]["apiServerAccessProfile"]["authorizedIPRanges"]:
-                                return CheckResult.PASSED
+                properties = conf.get('properties')
+                if not properties:
+                    return CheckResult.FAILED
+                if not isinstance(properties, DictNode):
+                    return CheckResult.FAILED
+                apiServerAccessProfile = properties.get('apiServerAccessProfile')
+                if not apiServerAccessProfile:
+                    return CheckResult.FAILED
+                authorizedIPRanges = apiServerAccessProfile.get('authorizedIPRanges')
+                if authorizedIPRanges:
+                    return CheckResult.PASSED
```

Comment:
@achiar99 What does this change fix? If the purpose is to refactor, then I think it's less readable then the original implementation

Reply:
i added tests..
in some cases `properties` is not a dict.
also i think this is more readable than nested ifs..